labelbox/schema/project.py

```diff
@@ -773,6 +779,156 @@ def create_batch(
+        but not both. When more than 100k data rows are specified and thus multiple batches are needed, the specific
+        batch that each data row will be placed in is undefined.
+
+        Batches will be created with the specified name prefix and a unique suffix. The suffix will be a 4-digit
+        number starting at 0000. For example, if the name prefix is "batch" and 3 batches are created, the names
+        will be "batch0000", "batch0001", and "batch0002". This method will throw an error if a batch with the same
+        name already exists.
+
+        Args:
+            name_prefix: a prefix for the batch names, must be unique within a project
+            data_rows: Either a list of `DataRows` or Data Row ids.
+            global_keys: global keys for data rows to add to the batch.
+            priority: An optional priority for the Data Rows in the Batch. 1 highest -> 5 lowest
+            consensus_settings: An optional dictionary with consensus settings: {'number_of_labels': 3,
+                'coverage_percentage': 0.1}
+
+        Returns: a task for the created batches
+        """
+
+        if self.queue_mode != QueueMode.Batch:
+            raise ValueError("Project must be in batch mode")
+
+        dr_ids = []
+        if data_rows is not None:
+            for dr in data_rows:
+                if isinstance(dr, Entity.DataRow):
+                    dr_ids.append(dr.uid)
+                elif isinstance(dr, str):
+                    dr_ids.append(dr)
+                else:
+                    raise ValueError(
+                        "`data_rows` must be DataRow ids or DataRow objects")
+
+        self._wait_until_data_rows_are_processed(
+            dr_ids, global_keys, self._wait_processing_max_seconds)
+
+        if consensus_settings:
+            consensus_settings = ConsensusSettings(**consensus_settings).dict(
+                by_alias=True)
+
+        method = 'createBatches'
+        mutation_str = """mutation %sPyApi($projectId: ID!, $input: CreateBatchesInput!) {
+                                      project(where: {id: $projectId}) {
+                                        %s(input: $input) {
+                                          tasks {
+                                            batchUuid
+                                            taskId
+                                          }
+                                        }
+                                      }
+                                    }
+                                """ % (method, method)
+
+        params = {
+            "projectId": self.uid,
+            "input": {
+                "batchNamePrefix": name_prefix,
+                "dataRowIds": dr_ids,
+                "globalKeys": global_keys,
+                "priority": priority,
+                "consensusSettings": consensus_settings
+            }
+        }
+
+        tasks = self.client.execute(
```

Comment:
what happens if mutation returns an error? I guess `client` will just throw?

Reply:
Yes, `client` will throw an error with the message that the backend services passed back